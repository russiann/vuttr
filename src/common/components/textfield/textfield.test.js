import React from 'react';
import {render, cleanup, fireEvent} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import Textfield from './textfield.component';

afterEach(cleanup);

it('renders', () => {
  const {asFragment} = render(<Textfield />);
  expect(asFragment()).toMatchSnapshot();
});

it('should render default value', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Textfield defaultValue={text} />);
  const input = getByTestId('textfield');
  expect(input.value).toBe(text);
});

it('should render typed text', () => {
  const text = 'Hello World!';
  const {getByTestId} = render(<Textfield />);
  const input = getByTestId('textfield');
  fireEvent.change(input, {target: {value: text}});
  expect(input.value).toBe(text);
});
