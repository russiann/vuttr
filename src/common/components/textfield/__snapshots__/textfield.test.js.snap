// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <input
    class="css-ufdmwz"
    data-testid="textfield"
    type="text"
  />
</DocumentFragment>
`;
